.github/workflows/main.yml

```diff
@@ -35,21 +35,26 @@ jobs:
     if: needs.changes.outputs.any_modified == 'true'
     runs-on: ubuntu-22.04
     timeout-minutes: 15
+    strategy:
+      matrix:
+        arch: [ "x86_64", "aarch64" ]
+        libc: [ "gnu", "musl" ]
     steps:
       - uses: actions/checkout@v3
       - name: Setup Rust toolchain and cache
         uses: actions-rust-lang/setup-rust-toolchain@v1
         with:
           components: rustfmt, clippy
-
       - name: typos-action
         uses: crate-ci/typos@v1.14.12
-
-      - uses: taiki-e/install-action@just
-
-      - name: Install requirements
-        run: sudo env PATH=$PATH just ci-prepare
-
+      - name: Install just
+        uses: taiki-e/install-action@just
+      - name: Install cross-rs
+        run: cargo install cross --git https://github.com/cross-rs/cross
+      - name: Setup target
+        run: |
+          echo "CARGO=cross" >> ${GITHUB_ENV}
+          echo "TARGET=${{ matrix.arch }}-unknown-linux-${{ matrix.libc }}" >> ${GITHUB_ENV}
       - name: Check formatting and lints
         run: just lint
```

Comment:
If I'm right, then I don't think we need matrix for lint checking, the `just lint` target should checks for all targets and features. Would that be affected by arch or libc type?

Reply:
The arch could make a difference with clippy when conditional compilation is used.

This also applies for dependencies using conditional compilation. An example of this is in `mock.rs` in this PR, where libc doesn't expose `__rlimit_resource_t` when building for musl.